examples/crab-saber/src/lib.rs

```diff
@@ -177,57 +177,54 @@ fn handle_state_change(
     game_context: &mut GameContext,
     world: &mut World,
 ) {
-    let mut objects_to_hide = Vec::new();
-    let mut objects_to_show = Vec::new();
-
     match (previous_state, current_state) {
         (SessionState::VISIBLE, SessionState::FOCUSED) => {
+            fn show(world: &mut World, entity: Entity) {
+                world.insert_one(entity, Visible {}).unwrap();
+            }
+
             audio_context.resume_music_track();
             match game_context.state {
                 GameState::Init => {}
                 GameState::MainMenu | GameState::GameOver => {
-                    objects_to_show.push(game_context.pointer);
+                    show(world, game_context.pointer);
                 }
                 GameState::Playing(_) => {
-                    objects_to_show.push(game_context.blue_saber);
-                    objects_to_show.push(game_context.red_saber);
+                    show(world, game_context.blue_saber);
+                    show(world, game_context.red_saber);
                 }
             }
         }
         (SessionState::FOCUSED, SessionState::VISIBLE) => {
+            fn hide(world: &mut World, entity: Entity) {
+                if world.remove_one::<Visible>(entity).is_err() {
+                    println!(
+                        "[STATE_CHANGE] Tried to make {:?} hidden but it had no Visible component",
+                        entity
+                    )
+                }
+            }
+
             audio_context.pause_music_track();
             match game_context.state {
                 GameState::Init => {}
                 GameState::MainMenu | GameState::GameOver => {
-                    objects_to_hide.push(game_context.pointer);
+                    hide(world, game_context.pointer);
                 }
                 GameState::Playing(_) => {
-                    objects_to_hide.push(game_context.blue_saber);
-                    objects_to_hide.push(game_context.red_saber);
+                    hide(world, game_context.blue_saber);
+                    hide(world, game_context.red_saber);
                 }
             }
         }
         _ => {}
     }
-
-    for e in objects_to_hide.drain(..) {
-        if world.remove_one::<Visible>(e).is_err() {
-            println!(
-                "[STATE_CHANGE] Tried to make {:?} hidden but it had no Visible component",
-                e
-            )
-        }
-    }
-
-    for e in objects_to_show.drain(..) {
-        world.insert_one(e, Visible {}).unwrap();
-    }
 }
 
-fn init(engine: &mut Engine) -> Result<(World, GameContext), HothamError> {
+fn init(engine: &mut Engine) -> (World, GameContext) {
```

Comment:
Oh, that's a good point! No need to wrap this in a result!

Just to keep things clean - do you think you could split this off into a separate PR? If not, no dramas, I can take care of it.